tensorizer/stream_io.py

```diff
@@ -162,13 +174,52 @@ def __init__(
 
         self._curl = None
         with _wide_pipes.widen_new_pipes():  # Widen on Windows
+            popen_start = time.time()
             self._curl = subprocess.Popen(
                 cmd,
                 stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
                 bufsize=256 * 1024 * 1024,
             )
+        popen_end = time.time()
 
         _wide_pipes.widen_pipe(self._curl.stdout.fileno())  # Widen on Linux
+        resp = self._curl.stderr.readline() # Block on the http header response
+        resp_begin = time.time()
+
+        # We reinitialize this object when seeking, so we don't want to overwrite
+        # these tracking variables if they already exist.
+        if not hasattr(self, "popen_latencies"):
+            self.popen_latencies: List[float] = []
+        if not hasattr(self, "http_response_latencies"):
+            self.http_response_latencies: List[float] = []
+        if not hasattr(self, "bytes_read"):
+            self.bytes_read: int = 0
+        if not hasattr(self, "bytes_skipped"):
+            self.bytes_skipped: int = 0
+        if not hasattr(self, "read_operations"):
+            self.read_operations: int = 0
+        self.popen_latencies.append(popen_end - popen_start)
+        self.http_response_latencies.append(resp_begin - popen_end)
+
+        if not resp.startswith(b"HTTP/1.1 2"):
```

Comment:
safety: this will break when we start returning HTTP/2 responses. leave a comment and fix later

Reply:
Already handled, as it blows up on the test cases.